src/policies/target_tracking_policy.R

```diff
@@ -0,0 +1,23 @@
+# The target tracking policy works with a scaling metric and only one target value. 
+# This policy focus more on delivering application availability than saving costs.
+target_tracking_policy <- function(system_utilization, policy_parameters, ...) {
+  
+  arguments <- list(...)
+  allocated <- arguments$allocated
+
+  target <- policy_parameters$target_value
+  lower_threshold <- target - policy_parameters$scale_in_threshold
+  vm_cores <- policy_parameters$vm_cores
+  adjustment = (arguments$demanded_cores / target - allocated) / vm_cores
```

Comment:
```suggestion
  adjustment <- (arguments$demanded_cores / target - allocated) / vm_cores
```

O nosso simulator lida com cores, então essa divisão por vm_cores não deve acontecer no cálculo do adjustment, uma vez que ele deve ser em cores e não em instâncias.

Para o scale-out (adição de cores), o correto aqui seria:
```R
adjustment <- (arguments$demanded_cores / target) - allocated
if (adjustment > 0) new_cores <- ceiling(adjustment/vm_cores) * vm_cores
```

Com isso eu pego o próximo múltiplo do número de cores das instâncias para adicionar.

Para o scale-in é o simétrico:

```R
else if (adjustment < 0) new_cores <- floor(adjustment/vm_cores) * vm_cores
else new_cores <- 0
```

Assim tu vai remover o múltiplo anterior do número de cores.

Reply:
Feito